### ISSUE TITLE
dlopen crashes application in mac arm 64

### ISSUE THREAD

**Issue by anoop4** [cisco/openh264#3622]
Hi,

I am using dlopen to load the v2.2.0 openh264 mac arm64 binary in my application and it crashes. The crash log points to the dlopen line.
is this a known issue? Are there any workarounds?

I see the same crash with v2.3.1 as well.

Thanks,
Srijith

**Comment by huili2** [comment]
No. It works for me. Not sure what's the root cause of your side.

**Comment by anoop4** [comment]
I see there is a reference to a hardcoded /usr/local/lib path in maca64 and maci64 binaries. This is affecting our application performance since disk cycles are utilized for searching /usr/local/lib as well. I would expect this to be an rpath instead of a hardcoded local path.

Can someone look into fixing this?

```
otool -D libopenh264-2.2.0-osx-arm64.6.dylib
libopenh264-2.2.0-osx-arm64.6.dylib:
/usr/local/lib/libopenh264.6.dylib
```

Trying to replace the rpath separately was what was causing the crash.
